opm/simulators/timestepping/AdaptiveTimeSteppingEbos.hpp

```diff
@@ -693,6 +682,21 @@ namespace Opm {
             assert(growthFactor_ >= 1.0);
         }
 
+        double timeUnitConversion(const UnitSystem& unitSystem, double value)
+        {
+            switch (unitSystem.getType()) {
+                case UnitSystem::UnitType::UNIT_TYPE_FIELD:
+                case UnitSystem::UnitType::UNIT_TYPE_METRIC:
+                    return value * unit::day;
+                    break;
+                case UnitSystem::UnitType::UNIT_TYPE_LAB:
+                    return value * unit::hour;
+                    break;
+                default:
+                    throw std::runtime_error("Unsupported unit type when initializing the time stepper");
+            }
```

Comment:
Is there a reason you did not write this as
```C++
return unitSystem.to_si(UnitSystem::time, value);
```

Reply:
I just copied @GitPaean code. Maybe he had a reason? If not I can update accordingly.

Reply:
There is no good reason. That function was not in my mind.  I think @bska's suggestion is good.

Reply:
It was not in my mind either. Thanks to @bska for pointing it out.